src/cdk-lib/amazonaurora/aurora-default-vector-store.ts

```diff
@@ -0,0 +1,257 @@
+import { buildCustomResourceProvider } from '../../common/helpers/custom-resource-provider-helper';
+import { generatePhysicalNameV2 } from '../../common/helpers/utils';
+import { buildVpc } from '../../common/helpers/vpc-helper';
+import { BedrockFoundationModel } from '../bedrock/models';
+
+
+export interface AmazonAuroraDefaultVectorStoreProps {
+  /**
+   * The embeddings model for the knowledge base.
+   * Must be identical as in the KnowledgeBaseV2 construct.
+   * This is due to the factor that the embeddings models
+   * have different vector dimensions and this construct
+   * needs to know the vector dimensions to create the vector
+   * index of appropriate dimensions in the Aurora database.
+   */
+  readonly embeddingsModel: BedrockFoundationModel;
+}
+
+/**
+  * Creates default AmazonAuroraVectorStore.
+  *
+  * It includes creation of a VPC with 3 subnets (public,
+  * private with NAT Gateway, private without NAT Gateway),
+  * with the Amazon Aurora Serverless V2 Cluster.
+  * The cluster has 1 writer/reader of PostgreSQL version 15.5
+  * instance (min capacity 0.5, max capacity 4). Lambda custom
+  * resource executes required pgvector and Amazon Bedrock Knowledge
+  * Base SQL queries against Aurora cluster
+  * during deployment. The secret containing databases credentials is
+  * being deployed and securely stored in AWS Secrets Manager.
+  * You must specify the same embeddings model that you used in
+  * KnowledgeBase construct.
+  * @see https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/AuroraPostgreSQL.VectorDB.html)
+  */
+export class AmazonAuroraDefaultVectorStore extends cdk.Resource {
+  /**
+   * The ARN of your Amazon Aurora DB cluster.
+   */
+  public readonly resourceArn: string;
+
+  /**
+   * The name of your Database.
+   */
+  public readonly databaseName: string;
+
+  /**
+   * The Table Name of your Amazon Aurora DB cluster.
+   */
+  public readonly tableName: string;
+
+  /**
+   * The Secret ARN of your Amazon Aurora DB cluster.
+   */
+  public readonly credentialsSecretArn: string;
+
+  /**
+   * Primary key of your Amazon Aurora DB cluster.
+   */
+  public readonly primaryKeyField: string;
+
+  /**
+   * Cluster identifier of your Amazon Aurora DB cluster.
+   */
+  public readonly clusterIdentifier: string;
+
+  /**
+   * Model used for embeddings.
+   */
+  public readonly embeddingsModel: BedrockFoundationModel;
+
+  /**
+   * An IAM policy that allows Data API access to Aurora.
+   * @private
+   */
+  private auroraPgCRPolicy: iam.ManagedPolicy;
+
+  constructor(
+    scope: Construct,
+    id: string,
+    props: AmazonAuroraDefaultVectorStoreProps,
+  ) {
+    super(scope, id);
+
+    this.databaseName = 'bedrock_vector_db';
+    this.tableName = 'bedrock_integration.bedrock_kb';
+    this.primaryKeyField = 'id';
+    this.clusterIdentifier = 'aurora-serverless-vector-cluster';
+    this.embeddingsModel = props.embeddingsModel;
+
+    const vpc = buildVpc(this, {});
```

Comment:
what if someone want to use their existingVPC? should we add one more prop in AmazonAuroraDefaultVectorStoreProps for existing VPC?

Reply:
Could be an option. The only issue that I have with that is that we need to need to be sure that VPC has a private and isolated subnets due to how I integrate Lambda that executed PGVector calls on Aurora Database (Aurora is in a private subnet and Lambda is in private isolated so it has Internet Gateway to be able to communicate with Aurora). We could add additional check and exceptions there to make sure user's vpc has these subnets included.

Reply:
Agree with Dinesh, that would give more flexibility for the end user. Maybe we could add that in a separate PR as we need to do some work on the helpers side